packages/instrumentor/instrument.ts

```diff
@@ -163,6 +175,25 @@ export class Instrumentor {
 			undefined;
 		return included && !excluded;
 	}
+
+	shouldCollectCodeCoverage(filepath: string): boolean {
+		const shouldCoverCustomHooks = this.shouldInstrument(
+			filepath,
+			false,
+			this.customHooks,
+			["nothing"]
+		);
+		return (
+			this.collectSourceCodeCoverage &&
+			(shouldCoverCustomHooks || this.shouldInstrument(filepath, false))
```

Comment:
I find the logic here a bit convoluted. I've pushed a commit to simplify it a bit. Could have a look?

Reply:
Much better! What is the advantage of using static methods here?

Reply:
This is to clarify that these methods do not need to access the object's `this` and thus can work on the class level. The performance gain, in this case, is relatively negligible.